fulp_modules/main_features/bloodsuckers/code/powers/feed.dm

```diff
@@ -139,38 +154,39 @@
 		feed_target = pick(targets_valid)//targets[1]
 		return TRUE
 
-/datum/action/bloodsucker/feed/ActivatePower()
-	// set waitfor = FALSE   <---- DONT DO THIS!We WANT this power to hold up Activate(), so Deactivate() can happen after.
+/datum/action/bloodsucker/feed/ActivatePower(mob/living/user = owner)
+//	set waitfor = FALSE   <---- DONT DO THIS! We WANT this power to hold up Activate(), so Deactivate() can happen after.
 	var/mob/living/target = feed_target // Stored during CheckCanUse(). Can be a grabbed OR adjecent character.
-	var/mob/living/user = owner
 	var/datum/antagonist/bloodsucker/bloodsuckerdatum = IS_BLOODSUCKER(user)
-	var/datum/antagonist/vassal/vassaldatum = IS_VASSAL(user)
 	// Am I SECRET or LOUD? It stays this way the whole time! I must END IT to try it the other way.
-	var/amSilent = (!target_grappled || owner.grab_state < GRAB_AGGRESSIVE && !bloodsuckerdatum.Frenzied) //  && iscarbon(target) // Non-carbons (animals) not passive. They go straight into aggressive.
-	// Initial Wait
-	var/feed_time = (amSilent ? 45 : 25) - (2.5 * level_current)
-	feed_time = max(15, feed_time)
-	if(bloodsuckerdatum && bloodsuckerdatum.Frenzied)
-		/// In a frenzy? No time to wait, drink blood NOW!
+	if(!bloodsuckerdatum?.Frenzied && (!target_grappled || owner.grab_state <= GRAB_PASSIVE)) // && iscarbon(target) // Non-carbons (animals) not passive. They go straight into aggressive.
+		amSilent = TRUE
+
+	if(bloodsuckerdatum?.Frenzied)
+		blood_take_mult = 2
 		feed_time = 5.5
+	else if(!amSilent)
+		blood_take_mult = 1
+		feed_time = 25 - (2.5 * level_current)
+	else
+		blood_take_mult = 0.3
+		feed_time = 45 - (2.5 * level_current)
+	feed_time = max(15, feed_time)
```

Comment:
this makes frenzied bloodsuckers still take at least 1.5 seconds to aggressively drain someone